pyomo/core/expr/visitor.py

```diff
@@ -1466,53 +1502,68 @@ def _expression_is_fixed(node):
 
-    def __init__(self, verbose, smap, compute_values):
+    _expression_handlers = None
+
+    def __init__(self, verbose, smap):
         super(_ToStringVisitor, self).__init__()
         self.verbose = verbose
         self.smap = smap
-        self.compute_values = compute_values
 
     def visit(self, node, values):
         """ Visit nodes that have been expanded """
-        tmp = []
+        if node.PRECEDENCE is None:
+            if self._expression_handlers \
+               and node.__class__ in self._expression_handlers:
+                return self._expression_handlers[node.__class__](
+                    self, node, values)
+            return node._to_string(values, self.verbose, self.smap)
+
         for i,val in enumerate(values):
             arg = node._args_[i]
 
             if arg is None:
-                tmp.append('Undefined')                 # TODO: coverage
+                values[i] = 'Undefined'  # TODO: coverage
             elif arg.__class__ in native_numeric_types:
-                tmp.append(val)
+                pass
             elif arg.__class__ in nonpyomo_leaf_types:
-                tmp.append("'{0}'".format(val))
+                values[i] = f"'{val}'"
             else:
                 parens = False
                 if not self.verbose and arg.is_expression_type():
-                    if node._precedence() < arg._precedence():
+                    if arg.PRECEDENCE is None:
+                        pass
+                    elif node.PRECEDENCE < arg.PRECEDENCE:
                         parens = True
-                    elif node._precedence() == arg._precedence():
+                    elif node.PRECEDENCE == arg.PRECEDENCE:
                         if i == 0:
-                            parens = node._associativity() != 1
+                            parens = node.ASSOCIATIVITY != LEFT_TO_RIGHT
                         elif i == len(node._args_)-1:
-                            parens = node._associativity() != -1
+                            parens = node.ASSOCIATIVITY != RIGHT_TO_LEFT
                         else:
                             parens = True
                 if parens:
-                    tmp.append("({0})".format(val))
-                else:
-                    tmp.append(val)
+                    values[i] = f"({val})"
+
+        if self._expression_handlers \
+           and node.__class__ in self._expression_handlers:
+            return self._expression_handlers[node.__class__](
+                self, node, values)
 
-        return node._to_string(tmp, self.verbose, self.smap, self.compute_values)
+        return node._to_string(values, self.verbose, self.smap)
 
     def visiting_potential_leaf(self, node):
         """
         Visiting a potential leaf.
 
         Return True if the node is not expanded.
         """
-        if node is None:
-            return True, None                           # TODO: coverage
+        if node is None:  # TODO: coverage
```

Comment:
Same question--codecov didn't mark as uncovered

Reply:
I guess not...